public/remote/js/controllers/mainController.js

```diff
@@ -160,6 +125,44 @@ app.controller("mainController", function($scope, $rootScope, config, $routePara
     $scope.setCurrentVideo = function(video){
         $scope.current.video = video;
 
+
+        // Load all related overlays
+        $overlayService.list_by_video($scope.current.video.video_id)
+        .then(function onSuccess(response) {
+            // Make sure the overlays are in the scenario
+            $scope.filter = {};
+            $scope.filter.relationship_type = "overlay";
+            $relationshipService.list_by_label("belongs_to", $scope.pagination, $scope.filter)
```

Comment:
Reasoning see below (duplicate `.catch()`)
```suggestion
            return $relationshipService.list_by_label("belongs_to", $scope.pagination, $scope.filter)
```